waku/v2/protocol/metadata/waku_metadata.go

```diff
@@ -219,30 +233,14 @@ func (wakuM *WakuMetadata) Connected(n network.Network, cc network.Conn) {
 
 		peerID := cc.RemotePeer()
 
-		logger := wakuM.log.With(logging.HostID("peerID", peerID))
-
-		shouldDisconnect := true
 		shard, err := wakuM.Request(wakuM.ctx, peerID)
-		if err == nil {
-			if shard == nil {
```

Comment:
Is this error handled by new changes?
Or is shard support something we are not worried about as of now?

Reply:
It is already handled by the changes. Instead of returning a null shard, I just return an error.